components/eventing-controller/controllers/subscription/beb/reconciler.go

```diff
@@ -98,199 +99,229 @@ func NewReconciler(ctx context.Context, client client.Client, applicationLister
-
-	result := ctrl.Result{}
-
-	// Ensure the object was not deleted in the meantime
-	if err := r.Client.Get(ctx, req.NamespacedName, actualSubscription); err != nil {
+	// fetch current subscription object and ensure the object was not deleted in the meantime
+	currentSubscription := &eventingv1alpha1.Subscription{}
+	if err := r.Client.Get(ctx, req.NamespacedName, currentSubscription); err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
-	// Handle only the new subscription
-	desiredSubscription := actualSubscription.DeepCopy()
 
-	// Bind fields to logger
-	log := utils.LoggerWithSubscription(r.namedLogger(), desiredSubscription)
+	// copy the subscription object, so we don't modify the source object
+	subscription := currentSubscription.DeepCopy()
 
-	// the APIRule for the desired subscription
-	var apiRule *apigatewayv1alpha1.APIRule
+	// bind fields to logger
+	log := utils.LoggerWithSubscription(r.namedLogger(), subscription)
+	log.Debugw("received new reconcile request")
 
-	if !isInDeletion(desiredSubscription) {
-		// ensure the finalizer is set
-		if err := r.syncFinalizer(ctx, desiredSubscription, &result, log); err != nil {
-			return ctrl.Result{}, errors.Wrap(err, "sync finalizer failed")
-		}
-		if result.Requeue {
-			return result, nil
-		}
+	// instantiate a return object
+	result := ctrl.Result{}
 
-		// sync the initial Subscription status
-		if statusChanged := r.syncInitialStatus(desiredSubscription); statusChanged {
-			if err := r.Status().Update(ctx, desiredSubscription); err != nil {
-				return ctrl.Result{}, errors.Wrap(err, "sync status failed")
-			}
-			return ctrl.Result{Requeue: true}, nil
-		}
+	// sync the initial Subscription status
+	r.syncInitialStatus(subscription)
 
-		// sync APIRule
-		var err error
-		apiRule, err = r.syncAPIRule(ctx, desiredSubscription, log)
-		if !recerrors.IsSkippable(err) {
-			return ctrl.Result{}, err
-		}
+	// handle deletion of the subscription
+	if isInDeletion(subscription) {
+		return r.handleDeleteSubscription(ctx, subscription, log)
+	}
 
-		// sync the Subscription status for the APIRule
-		statusChanged, err := r.syncSubscriptionAPIRuleStatus(actualSubscription, desiredSubscription, apiRule)
-		if err != nil {
-			return ctrl.Result{}, err
-		}
-		if statusChanged {
-			if err := r.Client.Status().Update(ctx, desiredSubscription); err != nil {
-				return ctrl.Result{}, err
-			}
-			return ctrl.Result{Requeue: true}, nil
+	// sync Finalizers, ensure the finalizer is set
+	if err := r.syncFinalizer(subscription, log); err != nil {
+		if updateErr := r.updateSubscription(ctx, subscription, log); updateErr != nil {
+			return ctrl.Result{}, errors.Wrap(err, updateErr.Error())
 		}
+		return ctrl.Result{}, errors.Wrap(err, "sync finalizer failed")
 	}
 
-	// mark if the subscription status was changed
-	statusChanged := false
+	// sync APIRule for the desired subscription
+	apiRule, err := r.syncAPIRule(ctx, subscription, log)
```

Comment:
One question: it seems in the code on main, the API rule logics happen earlier in the reconcile call. In your PR, they have been move down. Is that correct? If yes, why?

Reply:
No, We have followed the same sequence as in main:
1. syncFinalizer
2. syncAPIRule
3. syncBEBSubscription